config/v3_1_experimental/translate/translate.go

```diff
@@ -33,23 +33,23 @@ func translateFilesystem(old old_types.Filesystem) (ret types.Filesystem) {
 	return
 }
 
-func translateConfigReference(old old_types.ConfigReference) (ret types.ConfigReference) {
+func translateConfigReference(old old_types.ConfigReference) (ret types.Resource) {
```

Comment:
The elegance of this approach is definitely underscored by the fact that 3 separate types are now transparently converted to the same thing.

Reply:
:100:  I really like having the ability to add another resource easily.